rcsb/db/processors/DataTransformFactory.py

```diff
@@ -384,37 +383,54 @@ def castDateToObj(self, trfTup):
         if (origLength == 0) or (trfTup.value == "?") or (trfTup.value == "."):
             return TrfValue(self.__nullValueOther, trfTup.atId, origLength, True)
         tv = trfTup.value.replace(":", " ", 1)
-        return TrfValue(dateutil.parser.parse(tv), trfTup.atId, origLength, False)
+        # Note: Previously used dateutil.parse, which returns a datetime anyway.
+        stamp = datetime.fromisoformat(tv)
+        return TrfValue(stamp, trfTup.atId, origLength, False)
 
     def castDateTimeToIsoDate(self, trfTup):
-        """Cast the input date (optional time) string (yyyy-mm-dd:hh::mm:ss) to a Python DateTime object -
+        """
+        Fixes/modifies a TrfValue with value formatted as `yyyy-mm-dd[:hh::mm:ss]` **(note the colon)**.
+        Substitutes the first `:` with `T`, fills in `00:00:00`, and appends `+00:00`.
 
-        Return:  TrfValue tuple
+        Returns:
+            A TrfValue tuple
         """
+        # TODO
+        # What is this function doing???!
+        # Its name was castDateTimeToIsoDate, suggesting it outputs YYYY-MM-DD.
+        # Contradicting tht, the docstring said:
+        # > Cast the input date (optional time) string (yyyy-mm-dd:hh::mm:ss) to a Python DateTime object
+        # Instead, it did this:
+        # > tS = dateutil.parser.parse(tv).replace(tzinfo=pytz.UTC).isoformat()
+        # That's three contradictory claims.
+        # What was this function supposed to do?
+        # Why did it fill in 00:00:00, and why did it append +00:00?
+        # Also, why did the format separate the date and time with `:`?
+        # END TODO
```

Comment:
lol, these are all good questions. I imagine some of the contradictions probably came from John copying and pasting a docstring or changing the method name after already writing an [outdated] docstring. From experience, he wasn't always super careful with docstrings.

So, I think the best way to figure out what the intent of this was would be to run an actual load and add some logging to figure it out. I could look into doing that.